addons/account/models/account_reconcile_model.py

```diff
@@ -103,6 +126,30 @@ def _compute_float_amount(self):
             except ValueError:
                 record.amount = 0
 
+    @api.depends('model_id.counterpart_type')
+    def _compute_amount_type(self):
+        for line in self:
+            if line.model_id.counterpart_type in ('sale', 'purchase'):
+                line.amount_type = line.amount_type or 'percentage_st_line'
+            else:
+                line.amount_type = line.amount_type or 'percentage'
+
+    @api.depends('model_id.counterpart_type')
+    def _compute_journal_id(self):
+        for line in self:
+            if line.journal_id.type != line.model_id.counterpart_type:
+                line.journal_id = None
+            else:
+                line.journal_id = line.journal_id
```

Comment:
This doesn't seem correct in case of "journal entry". I think you should clear only in case of sale/purchase

Reply:
Before the current changes, the domain for the journal_id was `domain="[('type', '=', 'general')]"`. With the new domain `domain="[('type', '=', parent.counterpart_type)]"`, if someone creates the reco model with counterpart type "Vendor Bill" (domain for journal_id: type = 'purchase') and changes it to "Journal Entry", we should also clear the previous journal and allow only general journals to be selected